Documentation/contributors/design-proposals/subscription-status.md

```diff
@@ -0,0 +1,188 @@
+    UpToDate bool `json:"UpToDate"`
+
+    // LastUpdated represents the last time that the Subscription status was updated
+    LastUpdated metav1.Time       `json:"lastUpdated"`
+
+    // Conditions is a list of the latest available observations about a Subscription's current state.
+    Conditions []SubscriptionCondition
+}
+
+// SubscriptionConditionType indicates an explicit state condition about a Subscription in "abnormal-true"
+// polarity form (see https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties).
+type SusbcriptionConditionType string
+
+const (
+    // SubscriptionFailed indicates the Subscription has failed progressing the operator installation to the latest version.
+    SubscriptionFailed SubscriptionConditionType = "Failed"
+
+    // SubscriptionProgressing indicates the Subscription is attempting to progress the cluster to the desired state.
+    SubscriptionProgressing SubscriptionConditionType = "Progressing"
+
+    // SubscriptionCatalogSourcesUnhealthy indicates that some or all of the CatalogSources to be used in resolution are unhealthy.
+    SubscriptionCatalogSourcesUnhealthy SubscriptionConditionType = "CatalogSourcesUnhealthy"
+
+    // SubscriptionCatalogSourceInvalid indicates the CatalogSource specified in the SubscriptionSpec is not valid.
+    SubscriptionCatalogSourceInvalid SubscriptionConditionType = "CatalogSourceInvalid"
+
+    // SubscriptionPackageChannelInvalid indicates the package and channel specified in the SubscriptionSpec is not valid.
+    SubscriptionPackageChannelInvalid SubscriptionConditionType = "PackageChannelInvalid"
+
+    // SubscriptionInstallPlanFailed indicates the InstallPlan responsible for installing the current CSV has failed.
+    SubscriptionInstallPlanFailed SubscriptionConditionType = "InstallPlanFailed"
+
+    // SubscriptionInstallPlanMissing indicates the InstallPlan responsible for installing the current CSV is missing.
+    SubscriptionInstallPlanMissing SubscriptionConditionType = "InstallPlanMissing"
+
+    // SubscriptionInstallPlanAwaitingManualApproval indicates the InstallPlan responsible for installing the current CSV is waiting 
+    // for manual approval.
+    SubscriptionInstallPlanAwaitingManualApproval SubscriptionConditionType = "InstallPlanAwaitingManualApproval"
+
+    // SubscriptionInstalledCSVReplacementAvailable indicates there exists a replacement for the installed CSV.
+    SubscriptionInstalledCSVReplacementAvailable SubscriptionConditionType = "InstalledCSVReplacementAvailable"
+
+    // SubscriptionInstalledCSVMissing indicates the installed CSV is missing.
+    SubscriptionInstalledCSVMissing SubscriptionConditionType = "InstalledCSVMissing"
+
+    // SubscriptionInstalledCSVFailed indicates the installed CSV has failed.
+    SubscriptionInstalledCSVFailed SubscriptionConditionType = "InstalledCSVFailed"
```

Comment:
We can add that. I just want to keep the number of condition types low - since once we have a condition, it becomes painful to remove if OLM dependents rely on it.